lib/helper/Puppeteer.js

```diff
@@ -127,6 +128,22 @@ const consoleLogStore = new Console();
  * }
  * ```
  *
+* #### Example #5: Target URL with provided basic authentication
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Puppeteer : {
+ *        url: 'http://localhost',
+ *        basicAuth: 'username, password',
```

Comment:
What about passing an object instead of string?

Benefits of this would be no accidental trimming of white spaces as result of string manipulation, or handling of exceptions when password contains `,` and the I guess possibility of being able to use `secret` for password.
```js
basicAuth: {
  username: 'postman',
  password: secret('withtrailingspace ')
}

Reply:
I'm not sure the `secret` thingy, I'll check it out. Thanks man!

Reply:
with password, one possibility is passing it as env var. :)

Reply:
isn't it the case `secret` just help you mask printed string on log, with this config, you can still see it as plain?

Reply:
Yes, you are right, from the looks of it `secret` is shown as ['*****'](https://github.com/Codeception/CodeceptJS/blob/39225aebe7b03a44becfe0b58388a7ee9a90a92c/lib/step.js#L84-L85) in step log output.
With your proposed changes it actually removed the hardcoded `user:password@` auth string from the URL, as it is handled in the background by calling the [`authenticate`](https://github.com/GoogleChrome/puppeteer/blob/v1.20.0/docs/api.md#pageauthenticatecredentials) method directly on `Page` instance (which shows no log output as far I can recall).